package.py

```diff
@@ -0,0 +1,911 @@
+        return env
+
+    @staticmethod
+    def is_ci() -> bool:
+        return os.environ.get('CI') is not None
+
+    @staticmethod
+    def update_path(directory: Path) -> None:
+        path = os.environ.get('PATH')
+        os.environ.setdefault('PATH', f'{path};{directory}')
+
+
+class Checksum:
+    @staticmethod
+    def generate(env: Environment, path: Path) -> Path:
+        checksum_filename = f'{path.name}.sha512'
+        cmd = None
+        if env.is_mac():
+            cmd = f'shasum -a 512 {path} > {checksum_filename}'
+        elif env.is_linux():
+            cmd = f'sha512sum {path} > {checksum_filename}'
+        elif env.is_windows():
+            cmd = f'powershell.exe -command "Get-FileHash {path} -Algorithm SHA512 | Select-Object Hash | foreach {{$_.Hash}} | Out-File -FilePath {checksum_filename}"'  # noqa E501
+        else:
+            logger.error('unsupported system')
+            exit(1)
+
+        ret_code = subprocess.call(cmd, shell=True)
+        if ret_code != 0:
+            logger.error(f'could not generate sha512 sum for {path}')
+            exit(1)
+        return path.parent.joinpath(checksum_filename)
+
+
+class Storage:
+    def __init__(self) -> None:
+        self.working_directory = Path(os.getcwd())
+        self.dist_directory = self.working_directory.joinpath('dist')
+        self.build_directory = self.working_directory.joinpath('build')
+        self.wheel_directory = self.build_directory.joinpath('wheels')
+        self.temporary_directory = self.build_directory.joinpath('temp')
+        self.backend_directory = self.build_directory.joinpath('backend')
+        self.build_directory.mkdir(parents=True, exist_ok=True)
+
+    def prepare_backend(self) -> None:
+        if self.backend_directory.exists():
+            shutil.rmtree(self.backend_directory)
+        self.backend_directory.mkdir(parents=True, exist_ok=True)
+
+    def prepare_temp(self) -> None:
+        self.temporary_directory.mkdir(exist_ok=True, parents=True)
+
+    def move_to_dist(self, file: Path) -> None:
+        self.dist_directory.mkdir(exist_ok=True)
+        logger.info(f'moving {file.name} to {self.dist_directory}')
+        shutil.move(src=file, dst=self.dist_directory)
+
+    def check_backend(self) -> None:
+        backend = self.backend_directory
+        if not backend.exists() or not any(backend.iterdir()):
+            logger.error(f'{backend} was missing or empty')
+            exit(1)
+
+    def copy_to_dist(self, file: Path) -> None:
+        self.dist_directory.mkdir(exist_ok=True)
+        logger.info(f'copying {file.name} to {self.dist_directory}')
+        shutil.copy(src=file, dst=self.dist_directory)
+
+
+class WindowsPackaging:
+    def __init__(self, storage: Storage, env: Environment) -> None:
+        self.__storage = storage
+        self.__env = env
+
+    @log_group('miniupnpc windows')
+    def setup_miniupnpc(self) -> None:
+        miniupnc = 'miniupnpc_64bit_py39-2.2.24.zip'
```

Comment:
Do you thing that this should be moved to the class `__init__` and even be a class argument?

Reply:
No, because this is not used outside of the method itself. So there is no need to expose it to the outer scope.